opsdroid/connector/matrix/connector.py

```diff
@@ -142,6 +142,14 @@ def filter_json(self):
                 message = await self._parse_sync_response(response)
                 await self.opsdroid.parse(message)
 
+            except MatrixRequestError as mre:
+                # We can safely ignore timeout errors. The non-standard error
+                # codes are returned by Cloudflare.
+                if mre.code in [504, 522, 524]:
+                    _LOGGER.info('Matrix Sync Timeout (code: %d)', mre.code)
+                    continue
+
+                _LOGGER.exception('Matrix Sync Error')
```

Comment:
I'd remove this `_LOGGER` and change the above one from `.info` to `.exception` since it will be logged even if the level is set to info. This second log is a bit redundant as well since we already got the information that there was a Matris Sync Timeout

Reply:
I've set the logging of timeouts to the `INFO` level because they are pretty common and don't really cause any problems (ie. not really an error). They are also not really exceptions because they have been handled correctly. If I set my logging level at `ERROR` ([which includes `.exception()`](https://docs.python.org/3.7/library/logging.html#logging.Logger.exception)) then I really only want to know about errors that are potentially going to cause problems. There could be an argument made for logging them at the `WARNING` level, but `INFO` is just what I picked at the time.

The second `_LOGGER.exception('Matrix Sync Error')` is to report any `MatrixRequestError` exceptions that haven't been correctly handled (ie. errors that aren't timeouts). If we remove it then more serious errors won't be reported and will go unnoticed.

Reply:
That makes a lot of sense, thanks for taking the take to explain. I'm happy with the PR so I am going to update the branch and then merge this PR when travis is happy 😄 👍